src/components/PeopleList/PeopleList.tsx

```diff
@@ -0,0 +1,130 @@
+import classNames from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+
+type Props = {};
+
+export const PeopleList: React.FC<Props> = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+
+  const [isLoadingErrorShown, setIsLoadingErrorShown] = useState(false);
+
+  const { personSlug } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then((persons) => {
+        const preparedPeople: Person[] = persons.map((person: Person) => ({
+          ...person,
+          mother: persons.find(({ sex, name }: Person) => sex === 'f'
+            && name === person.motherName),
+          father: persons.find(({ sex, name }: Person) => sex === 'm'
+            && name === person.fatherName),
+        }));
+
+        setPeople(preparedPeople);
+      })
+      .catch(() => {
+        setIsLoadingErrorShown(true);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  return (
+    <div className="block">
+      <div className="box table-container">
+        {isLoading ? (
+          <Loader />
+        ) : (
+          <>
+            {(!isLoadingErrorShown && !!people.length) && (
```

Comment:
Conditions consisting of two or more parts should be placed in separate variables for better readability.